rpcserver.go

```diff
@@ -5322,8 +5322,10 @@ func (r *rpcServer) FeeReport(ctx context.Context,
 
 		// Self node should always have policies for its channels.
 		if edgePolicy == nil {
-			return fmt.Errorf("no policy for outgoing channel %v ",
-				chanInfo.ChannelID)
+			srvrLog.Warnf("No channel policy found for "+
```

Comment:
only warn if the channel is restored? otherwise return error?

Reply:
I wanted to do that, but we don't have the restored info at this level. We could retrieve it from the database though.

Reply:
maybe returning an error is the proper thing to do. none of the channel state is there, so how can you generate a fee report?

Reply:
Returning the error isn't ideal if you've restored a channel on a node with other live channels in it though, as the call will always fail.

Reply:
How often does that happen? is it even recommended?

Reply:
The flip side is that we will now silently succeed even w/o restored channels where we would have failed previously

Reply:
> How often does that happen? is it even recommended?

Probably not very often, but I see nothing wrong with the workflow.

> The flip side is that we will now silently succeed even w/o restored channels where we would have failed previously

Turns out we actually don't need to do anything where and can keep the error. `ForEachChannel` here is actually iterating over the edges stored in the graph, which wouldn't exist for restored channels, rather than the channel state. Ended up dropping the commit.

Reply:
> Probably not very often, but I see nothing wrong with the workflow.

If i found myself in this position, I would take the opportunity to make a fresh node as a chance to reevaluate how i ended up there and start from a clean slate. def can see the desire of people wanting to maintain node ids though

> Turns out we actually don't need to do anything where and can keep the error. ForEachChannel here is actually iterating over the edges stored in the graph, which wouldn't exist for restored channels, rather than the channel state. Ended up dropping the commit.

Heh didn't even realize that 😂